### PR TITLE
Reverse Linked List in C++

### PR DESCRIPTION
**Pull request by Abhijit2505**
### Description

Submission description here

Checklist:

- [x] I've checked the issue list before deciding what to submit.
- [x] I've read the contribution guidelines.
- [x] I've edited the `README.md` link to my code.

### CODE REVIEW

**Comment by Abhijit2505** [comment]
I have not edited any README.md file, please explain on that thing, I did not get it.

**Review by atarax665 — COMMENTED** [review]
Rename the file as Reverse_linked_list.cpp and Reverse_linked_list.py

File: C++/Linked List/ReverseLL.cpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
+#include <iostream>
+#include <cstdlib>
+#include <algorithm>
+#include <stack>
+
+using namespace std;
+
+/// Creating a class for Linked List object
```
Comment (atarax665) [inline C++/Linked List/ReverseLL.cpp]: Remove this whole comment.

File: C++/Linked List/ReverseLL.cpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    int data;
+    Node* next;
+    Node(int data)
+    {
+        this -> data = data;
+        this -> next = nullptr;
+    }
+};
+
+
```
Comment (atarax665) [inline C++/Linked List/ReverseLL.cpp]: Remove this extra line

File: C++/Linked List/ReverseLL.cpp [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            else
+            {
+                tail -> next = n;
+                tail = tail -> next;
+            }
+            cin >> data;
+        }
+        return head;
+    }
+
```
Comment (atarax665) [inline C++/Linked List/ReverseLL.cpp]: Remove this blank line too.

File: Python/Linked List/ReverseLL.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+# creating a solution class to implement more methods on the Linked List
+class solution:
+
+    def length(self,head):
+        temp = head
+        count = 0
+        while(temp):
+            temp = temp.next
+            count+=1
```
Comment (atarax665) [inline Python/Linked List/ReverseLL.py]: Add space before and after the operator.

**Comment by Abhijit2505** [comment]
> Rename the file as Reverse_linked_list.cpp and Reverse_linked_list.py

Made the changes ✔

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
@Abhijit2505 There is no need of creating a new folder with the name 'LinkedList'. You have to commit your code in the [ds folder](https://github.com/TesseractCoding/NeoAlgo/tree/master/C%2B%2B/ds)

**Comment by atarax665** [comment]
> I have not edited any README.md file, please explain on that thing, I did not get it.

You have to edit this [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/C%2B%2B/README.md) to link your program under the Data structure list. You have to do the same for the python file in the python folder.

File: C++/Linked List/Reverse_linked_list.cpp [hunk 5]
```diff
@@ -0,0 +1,159 @@
+#include <iostream>
+#include <cstdlib>
+#include <algorithm>
+#include <stack>
+
+using namespace std;
+
+/*
```
Comment (atarax665) [inline C++/Linked List/Reverse_linked_list.cpp]: Remove this whole comment, from line 8 to 17.

**Comment by Abhijit2505** [comment]
Three things,
* The space after count+= and the blank line you pointed out gets ignored by the compiler in C++ and Python when the compiled file is generated. There is no meaning of removing a blank line or adding some space. 

* ds folder is good, but there are around 40/50 Linked List algorithms which will be messy inside the ds folder. Hence I have created another folder named Linked List to make the repository modularized. Well as the number of program increases, you will understand the thing.

* Comments should be there inside every code to make it easier for some users. The comments which you have asked me to remover were not wrong neither had any faults, still as per your review I have removed them.

I was here just to know about the repository structure and mechanism because I am going to mentor the participants in the upcoming Contributor's Hack by HakinCodes.

Thank You.

**Comment by atarax665** [comment]
> Three things,
> 
> * The space after count+= and the blank line you pointed out gets ignored by the compiler in C++ and Python when the compiled file is generated. There is no meaning of removing a blank line or adding some space.
> * ds folder is good, but there are around 40/50 Linked List algorithms which will be messy inside the ds folder. Hence I have created another folder named Linked List to make the repository modularized. Well as the number of program increases, you will understand the thing.
> * Comments should be there inside every code to make it easier for some users. The comments which you have asked me to remover were not wrong neither had any faults, still as per your review I have removed them.
> 
> I was here just to know about the repository structure and mechanism because I am going to mentor the participants in the upcoming Contributor's Hack by HakinCodes.
> 
> Thank You.

All the points mentioned by you are quite legit but I have my own reasons for suggesting those changes.

Reasons:
* Yes, the space and blank lines get ignored by the compiler in C++ and Python when compiled file is generated, but proper use of spaces and blank lines enhance the readability of the code. [Refer to point 5 and 7 of this document](http://users.csc.calpoly.edu/~cstaley/General/CStyle.htm). Your code is well structured and that's commendable.

* We will be restructuring all the folders soon. For now we are merging all the commits to their respective domain folder. Making a new folder was justified from your end.

* I never said that the comment I asked you to remove was wrong or had any fault. I just felt that it was superfluous.

We are really excited about Contributor's Hack. Will love to work with you . Your codes are really well structured and are example for our other contributors. I hope I cleared my points.

Thank You

**Comment by atarax665** [comment]
@Abhijit2505 Can you please squash your commits so that I can merge this PR.

**Review by atarax665 — APPROVED** [review]

**Comment by Abhijit2505** [comment]
> @Abhijit2505 Can you please squash your commits so that I can merge this PR.

I think the Squash And Merge option is available for the people having write access to the repository. Hence please squash and merge it.

**Comment by atarax665** [comment]
@Abhijit2505 Okay, done.

**Comment by iamrajiv** [comment]
@all-contributors please add @Abhijit2505 for code

**Pull request merged**